multiversx_sdk_core/transaction_factories/token_transfers_data_builder.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any, List
+
+from multiversx_sdk_core.interfaces import IAddress
+from multiversx_sdk_core.serializer import arg_to_string, args_to_strings
+from multiversx_sdk_core.tokens import TokenComputer, TokenTransfer
+
+
+class TokenTransfersDataBuilder:
```

Comment:
Do you think we can re-design a bit and live without `TokenTransfersDataBuilder`, even if there would be a tiny little bit of duplication?

`if function` will only happen due to the needs of `SmartContractTransactionsFactory`.

I believe the amount of duplication, after removing this class (if it will be removed) will be negligible (and it won't be duplication per-se).

We can / should ask for other opinions.

However, we can also keep it, and also keep it out of specs.

Reply:
Re-designed it, and as we've discussed we will keep the builder as it very small and non-complex.